scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,441 @@
+#! /usr/bin/env python2
+"""
+usage: sonic_sku_create.py [-h] [-v] [-f FILE | -m MINIGRAPH] [-b BASE] [-r]
+                           [-c {new_sku_only,l2_mode_only,new_sku_l2}]
+                           [-s SKU] [-p] [-vv]
+
+Create a new SKU
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -v, --version         show program's version number and exit
+  -f FILE, --file FILE  SKU definition from xml file. -f OR -m must be provided when creating a new SKU
+  -m MINIGRAPH, --minigraph MINIGRAPH
+                        SKU definition from minigraph file. -f OR -m must be provided when creating a new SKU
+  -b BASE, --base BASE  SKU base definition  
+  -r, --remove          Remove SKU folder
+  -c {new_sku_only,l2_mode_only,new_sku_l2}, --cmd {new_sku_only,l2_mode_only,new_sku_l2}
+                        Choose action to preform (Generate a new SKU, Configure L2 mode, Both
+  -s SKU, --sku SKU     SKU name to be used when creating a new SKU or for  L2 configuration mode
+  -p, --print           Print port_config.ini without creating a new SKU
+  -vv, --verbose        Verbose output
+
+
+"""
+
+from __future__ import print_function
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import swsssdk
+import traceback
+import sys
+import shutil
+import sonic_platform
+
+from tabulate import tabulate
+from lxml import etree as ET
+from lxml.etree import QName
+
+minigraph_ns = "Microsoft.Search.Autopilot.Evolution"
+minigraph_ns1 = "http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution"
+
+DEFAULT_DEV_PATH = '/usr/share/sonic/device/'
+### port_config.ini header
+PORTCONFIG_HEADER = ["# name", "lanes", "speed", "alias", "index"]
+
+
+class SkuCreate(object):
+	"""
+	Process aclstat
+	"""
+
+	def __init__(self):
+
+		self.portconfig_dict = {}
+		self.platform_specific_dict = {"x86_64-mlnx_msn2700-r0":self.msn2700_specific}
+		self.default_lanes_per_port = []
+		self.fpp = []
+		self.fpp_split  =  {}
+		self.num_of_fpp = 0
+		self.num_of_lanes_dict = {"1" : 4, "2" : 2, "4" : 1}
+		
+
+		try:
+			platform = subprocess.check_output("sonic-cfggen -H -v DEVICE_METADATA.localhost.platform",shell=True) #self.metadata['platform']
+			self.platform = platform.rstrip()
+		except KeyError:
+			print ("Couldn't find platform info in CONFIG_DB DEVICE_METADATA", file=sys.stderr)
+			exit()
```

Comment:
>exit [](start = 3, length = 4)

In exception cases, exit with non-zero error codes. Multiple times in this script.